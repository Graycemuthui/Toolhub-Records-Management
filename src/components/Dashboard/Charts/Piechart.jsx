import React from "react";
import Chart from "chart.js/auto";
import { Pie } from "react-chartjs-2";
import { records } from "../../data/Mockdata";

const edited = records.filter((record) => record.edited === true);
const notEdited = records.filter((record) => record.edited === false);
const missing = records.filter((record) => record.age === "");
console.log(missing.length);
const labels = [
  "Total Records",
  "Non-Edited Records",
  "Edited Records",
  "Missing Records",
];
const data = {
  labels: labels,
  datasets: [
    {
      label: "My First dataset",
      borderColor: "0",
      height: "100px",
      backgroundColor: [
        "rgb(0,0,255)",
        "rgba(54, 162, 235, 1)",
        "rgba(255, 206, 86, 1)",
        "rgba(75, 192, 192, 1)",
      ],
      data: [records.length, notEdited.length, edited.length, missing.length],
      options: {
        responsive: true,
        maintainAspectRatio: false,
      },
    },
  ],
};

const Piechart = () => {
  return <Pie data={data} />;
};

export default Piechart;
