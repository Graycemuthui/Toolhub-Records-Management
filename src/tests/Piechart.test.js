import React from "react";
import TestRenderer from "react-test-renderer";
import Piechart from "../components/Dashboard/Charts/Piechart";

it("Piechart render correctly", () => {
  const tree = TestRenderer.create(<Piechart />).toJSON();
  expect(tree).toMatchSnapshot();
});
