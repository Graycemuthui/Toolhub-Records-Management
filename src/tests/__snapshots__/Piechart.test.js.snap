// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dashboard render correctly 1`] = `
<canvas
  height={150}
  role="img"
  width={300}
/>
`;

exports[`Piechart render correctly 1`] = `
<canvas
  height={150}
  role="img"
  width={300}
/>
`;
